### PR TITLE
ref(msteams): Use message builder for actions

### PR DESCRIPTION
**Pull request by 97varun**
Build issue actions using message builder.

* `build_group_actions` - Builds 3 actions. All of them have a common pattern - if toggled create a submit button that will undo the toggled action, otherwise they reveal a card with choices on how to perform the action. Extracted out this common pattern into `create_issue_action_block`
* Some temporary typing changes in `card_builder/base.py`
* Added some more blocks to `block.py` - `action_set_block`, `container_block`, `input_choice_set_block`

### CODE REVIEW

File: src/sentry/integrations/msteams/card_builder/__init__.py [hunk 1]
```diff
@@ -137,128 +149,105 @@ def build_group_footer(group: Group, rules: Sequence[Rule], event: Event) -> Col
...
+        event=event,
+        rules=rules,
+        integration=integration,
+        toggled=group.get_assignee(),
+        action=ACTION_TYPE.ASSIGN,
+        action_title=IssueConstants.ASSIGN,
+        reverse_action=ACTION_TYPE.UNASSIGN,
+        reverse_action_title=IssueConstants.UNASSIGN,
+        # card_kwargs
+        title=IssueConstants.ASSIGN_INPUT_TITLE,
```
Comment (97varun) [inline src/sentry/integrations/msteams/card_builder/__init__.py:238]: Added an extra title which is not present in the original card to make all of the actions consistent.

File: src/sentry/integrations/msteams/card_builder/__init__.py [hunk 2]
```diff
@@ -137,128 +149,105 @@ def build_group_footer(group: Group, rules: Sequence[Rule], event: Event) -> Col
...
+    integration: Integration,
+    toggled: bool,
+    action: ACTION_TYPE,
+    action_title: str,
+    reverse_action: ACTION_TYPE,
+    reverse_action_title: str,
+    **card_kwargs: Any,
+) -> Action:
+    if toggled:
+        data = generate_action_payload(reverse_action, event, rules, integration)
```
Comment (ceorourke) [inline src/sentry/integrations/msteams/card_builder/__init__.py:180]: What is `reverse_action` in this context? Like undoing something?

File: src/sentry/integrations/msteams/card_builder/__init__.py [hunk 3]
```diff
@@ -137,128 +149,105 @@ def build_group_footer(group: Group, rules: Sequence[Rule], event: Event) -> Col
...
-                        "type": "Action.Submit",
-                        "title": "Ignore",
-                        "data": generate_action_payload(
-                            ACTION_TYPE.IGNORE, event, rules, integration
-                        ),
-                    }
-                ],
-            },
-        }
+    ignore_action = create_issue_action_block(
```
Comment (ceorourke) [inline src/sentry/integrations/msteams/card_builder/__init__.py:211]: This is sooooo clean, I love it

**Review by ceorourke — APPROVED** [review]
Left one small question but it's non-blocking. This looks great!

File: src/sentry/integrations/msteams/card_builder/__init__.py (hunk 2 above)
Comment (97varun) [reply src/sentry/integrations/msteams/card_builder/__init__.py:180]: Yes, like action -> reverse_action -- ignore -> stop_ignoring, assingn -> unassign or resolve -> unresolve. This is the best name I could think of 😄. @ceorourke any suggestions for better names?

Comment (ceorourke) [reply src/sentry/integrations/msteams/card_builder/__init__.py:181]: Maybe `undo_action`? It's not bad as is, I just wanted to clarify.

Comment (97varun) [reply src/sentry/integrations/msteams/card_builder/__init__.py:181]: Oh I guess I'll keep `reverse_action` and add some comments.

**Pull request merged**